rpc/conn.c

```diff
@@ -1210,6 +1210,9 @@ static void rpc_conn_sessions_cleanup_fail(struct m0_rpc_conn *conn, bool fail)
 			m0_rpc_rcv_session_terminate(session);
 		}
 		m0_rpc_session_fini_locked(session);
+		if (!fail) {
```

Comment:
@JugalPatil  Why not free the session irrespective of 'fail' ?

Reply:
This is a partial fix for rpc session.  Without the above condition, we are hitting segfault. 
so the above condition is required.

Reply:
I have the same question with @nkommuri .
The session is already finalised in `m0_rpc_session_fini_locked(session);` and it is removed from the list.
When shall we free this session? It is still leaked if `fail` is true?